### ISSUE TITLE
Enigma: Error on new key-pair generation "Unable to import key(s)! Internal error."

### ISSUE THREAD

**Issue by lutel** [roundcube/roundcubemail#5272]
errors
[24-May-2016 15:20:08 +0200]: <99r3f1e6> PHP Error: Enigma plugin: Unknown error importing GPG key. Please use the 'debug' option when creating the Crypt_GPG object, and file a bug report at http://pear.php.net/bugs/report.php?package=Crypt_GPG in /var/www/localhost/htdocs/roundcube/plugins/enigma/lib/enigma_engine.php on line 1069 (POST /?_task=settings&_framed=1&_action=plugin.enigmakeys) 

enigma
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: BEGIN PROCESSING
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: selecting streams
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => got 1
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG is ready for input
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to write 3663 bytes to GPG input
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => wrote 3663 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => closing GPG input pipe
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: selecting streams
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => got 2
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG error stream ready for reading
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to read 8192 bytes from GPG error
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => read 584 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31: secret key imported
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31 was created 3 seconds in the future (time warp or clock problem)
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31 was created 3 seconds in the future (time warp or clock problem)
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31 was created 3 seconds in the future (time warp or clock problem)
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31 was created 3 seconds in the future (time warp or clock problem)
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31: no valid user IDs
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: this may be caused by a missing self-signature
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: Total number processed: 1
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: w/o user IDs: 1
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: secret keys read: 1
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: secret keys imported: 1
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG status stream ready for reading
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to read 8192 bytes from GPG status
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => read 111 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: STATUS: IMPORT_OK 17 B7554EA9F293969B85563260424C2DCFCDD30A31
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: STATUS: IMPORT_RES 1 1 0 0 0 0 0 0 0 1 1 0 0 0
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => closing GPG input pipe
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: selecting streams
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => got 3
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG output stream ready for reading
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to read 8192 bytes from GPG output
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => read 0 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG error stream ready for reading
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to read 8192 bytes from GPG error
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => read 0 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG status stream ready for reading
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to read 8192 bytes from GPG status
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => read 0 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => closing GPG input pipe
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: END PROCESSING
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: CLOSING GPG SUBPROCESS
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => subprocess returned an unexpected exit code: 2
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: STOPPING GPG-AGENT DAEMON
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG-AGENT DAEMON STOPPED

After several attempts it creates the key pair, however it never accepts the passphrase when signing email.

**Comment by lutel** [comment]
Actually it doesn't work with 2.0.30 neither, 

errors
`[24-May-2016 15:20:08 +0200]: <99r3f1e6> PHP Error: Enigma plugin: Unknown error importing GPG key. Please use the 'debug' option when creating the Crypt_GPG object, and file a bug report at http://pear.php.net/bugs/report.php?package=Crypt_GPG in /var/www/localhost/htdocs/roundcube/plugins/enigma/lib/enigma_engine.php on line 1069 (POST /?_task=settings&_framed=1&_action=plugin.enigmakeys)
`

enigma
`[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: BEGIN PROCESSING
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: selecting streams
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => got 1
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG is ready for input
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to write 3663 bytes to GPG input
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => wrote 3663 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => closing GPG input pipe
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: selecting streams
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => got 2
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG error stream ready for reading
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to read 8192 bytes from GPG error
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => read 584 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31: secret key imported
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31 was created 3 seconds in the future (time warp or clock problem)
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31 was created 3 seconds in the future (time warp or clock problem)
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31 was created 3 seconds in the future (time warp or clock problem)
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31 was created 3 seconds in the future (time warp or clock problem)
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: key CDD30A31: no valid user IDs
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: this may be caused by a missing self-signature
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg: Total number processed: 1
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg:           w/o user IDs: 1
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg:       secret keys read: 1
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: ERROR: gpg:   secret keys imported: 1
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG status stream ready for reading
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to read 8192 bytes from GPG status
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => read 111 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: STATUS: IMPORT_OK 17 B7554EA9F293969B85563260424C2DCFCDD30A31
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: STATUS: IMPORT_RES 1 1 0 0 0 0 0 0 0 1 1 0 0 0
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => closing GPG input pipe
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: selecting streams
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => got 3
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG output stream ready for reading
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to read 8192 bytes from GPG output
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => read 0 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG error stream ready for reading
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to read 8192 bytes from GPG error
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => read 0 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG status stream ready for reading
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => about to read 8192 bytes from GPG status
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => read 0 bytes
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => closing GPG input pipe
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: END PROCESSING
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: CLOSING GPG SUBPROCESS
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: => subprocess returned an unexpected exit code: 2
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: STOPPING GPG-AGENT DAEMON
[24-May-2016 15:21:25 +0200]: <99r3f1e6> GPG: GPG-AGENT DAEMON STOPPED`

**Comment by himbeere** [comment]
Hello.

Same error here:

```
[25-May-2016 09:09:27 +0200]: <2logopoh> PHP Error: Enigma plugin: Unknown error importing GPG key. Please use the 'debug' option when creating the Crypt_GPG object, and file a bug report at http://pear.php.net/bugs/report.php?package=Crypt_GPG in /var/www/webmail/htdocs/plugins/enigma/lib/enigma_engine.php on line 1069 (POST /?_task=settings&_framed=1&_action=plugin.enigmakeys)
```

**Comment by himbeere** [comment]
Here is the error message:

`
[25-May-2016 15:21:52 Europe/Berlin] PHP Fatal error:  Uncaught Error: Call to undefined method Crypt_GPG_SubKey::usage() in /var/www/webmail.meine-oma.de/htdocs/plugins/enigma/lib/enigma_driver_gnupg.php:437
Stack trace:
#0 /var/www/webmail.meine-oma.de/htdocs/plugins/enigma/lib/enigma_driver_gnupg.php(221): enigma_driver_gnupg->parse_key(Object(Crypt_GPG_Key))
#1 /var/www/webmail.meine-oma.de/htdocs/plugins/enigma/lib/enigma_engine.php(938): enigma_driver_gnupg->list_keys(NULL)
#2 /var/www/webmail.meine-oma.de/htdocs/plugins/enigma/lib/enigma_ui.php(243): enigma_engine->list_keys(NULL)
#3 /var/www/webmail.meine-oma.de/htdocs/plugins/enigma/lib/enigma_ui.php(76): enigma_ui->key_list()
#4 /var/www/webmail.meine-oma.de/htdocs/plugins/enigma/enigma.php(417): enigma_ui->init()
#5 /var/www/webmail.meine-oma.de/htdocs/program/lib/Roundcube/rcube_plugin_api.php(491): enigma->preferences_ui()
#6 /var/www/webmail.meine-oma.de/htdocs/index.php(277): rcube_plugin_api->exec_action('plugin.enigmake...')
#7 {main}
  thrown in /var/www/webmail.meine-oma.de/htdocs/plugins/enigma/lib/enigma_driver_gnupg.php on line 437
`

**Comment by alecpl** [comment]
@himbeere  This fatal error says you don't use Crypt_GPG 1.4.1.
@lutel Passphrase maybe another issue, check your Roundcube and http server logs. As for the import issue. It complains about a few things in the key. Check time on your server and client. What gnupg version?

**Comment by himbeere** [comment]
@alecpl I solved my problem by doing a complete fresh install of roundcube. I had an old subkey.php in the roundcubemailfolder. thanks.

**Comment by lutel** [comment]
@alecpl time difference (between server and workstation-client) is within 5s. The log entry seems strange to me as i was expecting all key pair generation is done on server side, I don't know where time difference mentioned in log comes from. I've checked gnupg versions 2.0.28 to 2.0.30. I can't reproduce passphrase problem as I'm not able to generate keypair anymore. I could check it only once when somehow after several attempts key pair has been generated.

**Comment by alecpl** [comment]
Keys are generated client-side if browser supports that, then they are just imported as any other keys. Generating keys on server is problematic as it hits entropy issues, better to disable it. What browser and version?

**Comment by lutel** [comment]
@alecpl Chrome 51 on Mac, i've sync time between client and server, the issue still remains:

[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: /usr/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options --always-trust --homedir '/var/www/localhost/htdocs/roundcube/plugins/enigma/home/txx@xx.pl' --version
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: BEGIN PROCESSING
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: selecting streams
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => got 1
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG output stream ready for reading
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG output
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => read 647 bytes
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: selecting streams
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => got 3
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG output stream ready for reading
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG output
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => read 0 bytes
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG error stream ready for reading
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG error
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => read 0 bytes
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG status stream ready for reading
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG status
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => read 0 bytes
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: END PROCESSING
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: CLOSING GPG SUBPROCESS
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: USING GPG 2.0.28 with PHP 5.6.20-pl0-gentoo
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: OPENING GPG-AGENT SUBPROCESS WITH THE FOLLOWING COMMAND:
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: /usr/bin/gpg-agent --daemon --options /dev/null --csh --keep-display --no-grab --ignore-cache-for-signing --pinentry-touch-file /dev/null --disable-scdaemon --no-use-standard-socket --pinentry-program '/var/www/localhost/htdocs/roundcube/vendor/pear-pear.php.net/Crypt_GPG/Crypt/GPG/../../bin/crypt-gpg-pinentry' --homedir '/var/www/localhost/htdocs/roundcube/plugins/enigma/home/xx@xx.pl'
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: CLOSING GPG-AGENT LAUNCH PROCESS
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG-AGENT LAUNCH PROCESS CLOSED
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: /usr/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options --no-permission-warning --exit-on-status-write-error --trust-model always --homedir '/var/www/localhost/htdocs/roundcube/plugins/enigma/home/xx@xx.pl' --import
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: BEGIN PROCESSING
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: selecting streams
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => got 1
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG is ready for input
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => about to write 3655 bytes to GPG input
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => wrote 3655 bytes
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => closing GPG input pipe
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: selecting streams
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => got 1
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG error stream ready for reading
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG error
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => read 116 bytes
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: ERROR: gpg: key FF0A6901: secret key imported
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: ERROR: gpg: assuming bad signature from key FF0A6901 due to an unknown critical bit
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => closing GPG input pipe
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: selecting streams
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => got 3
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG output stream ready for reading
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG output
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => read 0 bytes
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG error stream ready for reading
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG error
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => read 290 bytes
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: ERROR: gpg: assuming bad signature from key FF0A6901 due to an unknown critical bit
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: ERROR: gpg: key FF0A6901: no valid user IDs
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: ERROR: gpg: this may be caused by a missing self-signature
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: ERROR: gpg: Total number processed: 1
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: ERROR: gpg:           w/o user IDs: 1
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: ERROR: gpg:       secret keys read: 1
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: ERROR: gpg:   secret keys imported: 1
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG status stream ready for reading
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG status
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => read 111 bytes
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: STATUS: IMPORT_OK 17 91FABCEE268FE9727BC116A43F4D44C4FF0A6901
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: STATUS: IMPORT_RES 1 1 0 0 0 0 0 0 0 1 1 0 0 0
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => closing GPG input pipe
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: selecting streams
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => got 2
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG error stream ready for reading
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG error
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => read 0 bytes
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG status stream ready for reading
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG status
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => read 0 bytes
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => closing GPG input pipe
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: END PROCESSING
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: CLOSING GPG SUBPROCESS
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: => subprocess returned an unexpected exit code: 2
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: STOPPING GPG-AGENT DAEMON
[26-May-2016 15:53:22 +0200]: <j22catqb> GPG: GPG-AGENT DAEMON STOPPED

**Comment by alecpl** [comment]
I tested with Chromium 50 and GnuPG 2.0.22 and import works here:
```
GPG: STATUS: IMPORT_OK 17 A09BBE3E8C1DE521DF760DF16D5C7CD9FAAFF515
GPG: STATUS: IMPORTED 6D5C7CD9FAAFF515 test <some@mail.pl>
GPG: STATUS: IMPORT_OK 1 A09BBE3E8C1DE521DF760DF16D5C7CD9FAAFF515
GPG: STATUS: IMPORT_RES 1 0 1 1 0 0 0 0 0 1 1 0 0 0
```
There are no "no valid user IDs ... this may be caused by a missing self-signature" errors. So, my question is: What do you have in the key generation form exactly? From the other side, it already says IMPORT_OK, but exits with code 2, which is unexpected.

**Comment by alecpl** [comment]
I just found that the import command may require --yes argument. Could you check if adding
```
$arguments[] = '--yes';
```
in line 1634 of /vendor/pear-pear.php.net/Crypt_GPG/Crypt/GPG.php file helps?

**Comment by lutel** [comment]
Adding --yes argument didn't help. In the form I see combos to choose identity, key size and password. After I click "save" I see "generating keys" message and 1-2 seconds after that the error in topic. Here again is the snapshot from enigma log after adding --yes....

`27-May-2016 19:49:06 +0200]: <j22catqb> GPG: /usr/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options --no-permission-warning --exit-on-status-write-error --trust-model always --yes --homedir '/var/www/localhost/htdocs/roundcube/plugins/enigma/home/xxx@xxx.pl' --import
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: BEGIN PROCESSING
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: selecting streams
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => got 1
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: GPG is ready for input
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => about to write 3655 bytes to GPG input
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => wrote 3655 bytes
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => closing GPG input pipe
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: selecting streams
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => got 2
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: GPG error stream ready for reading
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG error
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => read 406 bytes
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: ERROR: gpg: key 95F36161: secret key imported
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: ERROR: gpg: assuming bad signature from key 95F36161 due to an unknown critical bit
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: ERROR: gpg: assuming bad signature from key 95F36161 due to an unknown critical bit
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: ERROR: gpg: key 95F36161: no valid user IDs
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: ERROR: gpg: this may be caused by a missing self-signature
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: ERROR: gpg: Total number processed: 1
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: ERROR: gpg:           w/o user IDs: 1
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: ERROR: gpg:       secret keys read: 1
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: ERROR: gpg:   secret keys imported: 1
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: GPG status stream ready for reading
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG status
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => read 111 bytes
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: STATUS: IMPORT_OK 17 F9BB9A4466AA1D3DC428696E55E63C5395F36161
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: STATUS: IMPORT_RES 1 1 0 0 0 0 0 0 0 1 1 0 0 0
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => closing GPG input pipe
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: selecting streams
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => got 3
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: GPG output stream ready for reading
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG output
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => read 0 bytes
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: GPG error stream ready for reading
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG error
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => read 0 bytes
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: GPG status stream ready for reading
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => about to read 8192 bytes from GPG status
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => read 0 bytes
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => closing GPG input pipe
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: END PROCESSING
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: CLOSING GPG SUBPROCESS
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: => subprocess returned an unexpected exit code: 2
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: STOPPING GPG-AGENT DAEMON
[27-May-2016 19:49:06 +0200]: <j22catqb> GPG: GPG-AGENT DAEMON STOPPED`

**Comment by er47** [comment]
Some problem here. Exactly same messages.
I tried `$arguments[] = '--yes';` as well, but no success.

**Comment by alecpl** [comment]
Could you guys extract the key body from the http request and then call import command with the same arguments (skipping --status-fd and --command-fd arguments), to see what exactly gnupg displays. I guess it expects some input from the user, but that's just a guess, would be good to be sure. Call the command as the apache (www-data, etc.) user to also catch possible permission issues.

**Comment by er47** [comment]
Thanks for the hint. It is working now. In my case it was a time problem. ntpd was not running

**Comment by tsenger** [comment]
@himbeere I had the same problem as described here. I had updated to Roundcube 1.2.0 from an earlier version. The solution for me was, to delete the folder "Crypt" which was inside the folder /plugins/enigma/lib/ 
After removing this directory everything works fine.

**Comment by himbeere** [comment]
@tsenger Yeah i did that too. Maybe the roundcubemail guys should mention that in the upgrade guide.

cheers

**Comment by lutel** [comment]
@himbeere @tsenger it didn't work for me, i have clean 1.2.0 installation without "Crypt" folder

**Comment by petamem** [comment]
Same problem here, clean installation without Crypt folder, client and server running NTP

**Comment by CRtEurope** [comment]
@petamem Isn't gpg (GnuPG) 2.1.12 the wrong version? 2.1.12ersion 2.1.12 should not work. It is incompatible.

**Comment by petamem** [comment]
hm. downgraded to gpg 2.0.30, restarted gpg

gpg-agent --daemon
GPG_AGENT_INFO=/tmp/gpg-KCljwk/S.gpg-agent:6256:1; export GPG_AGENT_INFO;

still the same when trying to generate a key pair. restarted Apache. No error in logs.

**Comment by xm74** [comment]
The same thing here.
`root@beta:/usr/local/www/roundcube/plugins/enigma # uname -v
FreeBSD 10.3-RELEASE-p4 #0: Sat May 28 12:23:44 UTC 2016     root@amd64-builder.daemonology.net:/usr/obj/usr/src/sys/GENERIC
root@beta:/usr/local/www/roundcube/plugins/enigma # gpg -h | head -n 2
gpg (GnuPG) 2.0.29
libgcrypt 1.7.0
`

While trying to generate keys get this error

`root@beta:/usr/local/www/roundcube/plugins/enigma # tail ../../logs/errors
[10-Jun-2016 13:40:39 +0200]: <337304d1> PHP Error: Enigma plugin: Unknown error importing GPG key. Please use the 'debug' option when creating the Crypt_GPG object, and file a bug report at http://pear.php.net/bugs/report.php?package=Crypt_GPG in /usr/local/www/roundcube/plugins/enigma/lib/enigma_engine.php on line 1106 (POST /?_task=settings&_framed=1&_action=plugin.enigmakeys`

Debug log

`root@beta:/usr/local/www/roundcube/plugins/enigma # cat ../../logs/enigma
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: /usr/local/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options --always-trust --homedir '/usr/local/www/roundcube/plugins/enigma/home/max@kostikov.co' --version
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: BEGIN PROCESSING
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: selecting streams
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => got 1
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG output stream ready for reading
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => about to read 8192 bytes from GPG output
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => read 622 bytes
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: selecting streams
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => got 3
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG output stream ready for reading
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => about to read 8192 bytes from GPG output
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => read 0 bytes
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG error stream ready for reading
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => about to read 8192 bytes from GPG error
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => read 0 bytes
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG status stream ready for reading
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => about to read 8192 bytes from GPG status
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => read 0 bytes
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: END PROCESSING
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: CLOSING GPG SUBPROCESS
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: USING GPG 2.0.29 with PHP 5.6.22
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: OPENING GPG-AGENT SUBPROCESS WITH THE FOLLOWING COMMAND:
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: /usr/local/bin/gpg-agent --daemon --options /dev/null --csh --keep-display --no-grab --ignore-cache-for-signing --pinentry-touch-file /dev/null --disable-scdaemon --no-use-standard-socket --pinentry-program '/usr/local/www/roundcube/vendor/pear-pear.php.net/Crypt_GPG/Crypt/GPG/../../bin/crypt-gpg-pinentry' --homedir '/usr/local/www/roundcube/plugins/enigma/home/max@kostikov.co'
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: CLOSING GPG-AGENT LAUNCH PROCESS
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG-AGENT LAUNCH PROCESS CLOSED
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: /usr/local/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options --no-permission-warning --exit-on-status-write-error --trust-model always --homedir '/usr/local/www/roundcube/plugins/enigma/home/max@kostikov.co' --import
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: BEGIN PROCESSING
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: selecting streams
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => got 1
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG is ready for input
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => about to write 3631 bytes to GPG input
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => wrote 3631 bytes
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => closing GPG input pipe
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: selecting streams
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => got 1
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG error stream ready for reading
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => about to read 8192 bytes from GPG error
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => read 588 bytes
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: ERROR: gpg: key 0D20E76A: secret key imported
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: ERROR: gpg: key 0D20E76A was created 31 seconds in the future (time warp or clock problem)
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: ERROR: gpg: key 0D20E76A was created 31 seconds in the future (time warp or clock problem)
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: ERROR: gpg: key 0D20E76A was created 31 seconds in the future (time warp or clock problem)
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: ERROR: gpg: key 0D20E76A was created 31 seconds in the future (time warp or clock problem)
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: ERROR: gpg: key 0D20E76A: no valid user IDs
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: ERROR: gpg: this may be caused by a missing self-signature
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: ERROR: gpg: Total number processed: 1
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: ERROR: gpg:           w/o user IDs: 1
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: ERROR: gpg:       secret keys read: 1
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: ERROR: gpg:   secret keys imported: 1
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => closing GPG input pipe
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: selecting streams
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => got 3
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG output stream ready for reading
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => about to read 8192 bytes from GPG output
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => read 0 bytes
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG error stream ready for reading
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => about to read 8192 bytes from GPG error
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => read 0 bytes
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG status stream ready for reading
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => about to read 8192 bytes from GPG status
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => read 111 bytes
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: STATUS: IMPORT_OK 17 3DDABF41C0AA422971DA1258C47F2A380D20E76A
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: STATUS: IMPORT_RES 1 1 0 0 0 0 0 0 0 1 1 0 0 0
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => closing GPG input pipe
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: selecting streams
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => got 1
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG status stream ready for reading
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => about to read 8192 bytes from GPG status
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => read 0 bytes
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => closing GPG input pipe
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: END PROCESSING
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: CLOSING GPG SUBPROCESS
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: => subprocess returned an unexpected exit code: 2
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: STOPPING GPG-AGENT DAEMON
[10-Jun-2016 13:40:39 +0200]: <337304d1> GPG: GPG-AGENT DAEMON STOPPED`

But keys are persist in _/home_ directory

`root@beta:/usr/local/www/roundcube/plugins/enigma # ls -R home/
max@kostikov.co

home/max@kostikov.co:
private-keys-v1.d       pubring.gpg             secring.gpg

home/max@kostikov.co/private-keys-v1.d:`

What I'm doing wrong?

**Comment by github-johnny** [comment]
@lutel I've been following this thread since its creation.  Probably an hour or so after...
Sorry for the book.  tldr; pretty sure it's a hardware issue.

I have the same problem you do.  I think what you and I are experiencing is different than what some of the others here are referring to, ie.:  ntp issues, configuration issues.

I think we're rocking an old fashioned hardware issue.  Not on your servers side, but your workstation you are generating the keys on.  If you could, please try to generate the keys on a completely different type of machine.  It doesn't seem to matter what the OS or browser is.

The day I upgraded to 1.2.0 I found that I could import keys, but could not create keys.  I would end up with files created with a 0 byte size.  I would get logs identical to yours.  I initially thought it was a time issue.  After screwing with ntp and finally flat out doing it by hand and reboots and yada yada, I decided that time is not the problem and put it back how I had it originally.  It was getting late, I was tired of screwing with it.

The next morning when I woke up, I had this crazy idea that I changed a configuration setting and forgot to put it back.  I jumped on a Mac and tested it, it worked.  I thought that was great, ran downstairs to work on the main box like I had been.  It failed.

I verified the server settings and I had them all correct, I was wrong when I woke up.  They were all they way they should have been.  And how they should have been, it does work.

I don't want to give out too much of how my server environment is, but ours are pretty similar.  I found that running the key generation server side does not make a difference, but I'm also not 100% sold that I set that part up correctly.

Once I figured out that the configuration was correct and working, I then moved off of my main box and started to test other machines.  This is what I found:

OS is not a part of it.  I run Linux.  My main box wouldn't make the key, I thought it could be an OS issue.  I fired up a Windows VM and tested.  It too failed in the exact same way.  Then I thought, maybe a browser?  I tested IE, Firefox, and Chrome.  All the exact same results.  Whether in Linux or Windows.

I went to a different machine, completely different hardware, but running the exact same version of Linux as the main box.  Tested there, worked correctly.

The Mac I tested further and all browsers were fine.

This rules out OS issues, browsers, and configuration / Server issues.

It looks like it all points to the physical hardware that the machine is using.  My guess is either the Proc or the Mobo, probably the Proc.  I don't have a TPM chip or any of the Intel AES acceleration NSA spying goodness on the main machines proc.  It was very hard to get enough entropy to do key generation.

Sorry if this is formatted poorly or just completely wrong for this environment.  I've tried really hard to not have to post this, but it doesn't seem like sitting on my hands is really donig anyone any good.

Please let me know if you have questions.

@alecpl  Mighty fine work, sir.  This Enigma plugin has come a very long way.  I've been using it since the 0.8.x series, maybe even earlier, to be able to read messages.  The upgrades are very impressive.  Thank you for your hard work.

**Comment by alecpl** [comment]
All-in-all to generate keys client-side we're using openpgp.js library written in javascript. So, it's also possible the bug is there or is browser-specific or browser-version-specific or OS-specific or....

That's why I asked (did I?) to see how the key looks like (you should see it in http request in browser console). Not that I'll be able to say anything about it, but there might be someone with more in-deep knowledge about PGP encryption.

It's also not clear if this also fails with older versions of GnuPG. Does it? Maybe newest versions are more "fragile" to the imported key content or sth.

**Comment by hestefan** [comment]
I can't import my existing keys.
Roundcube 1.2.0
gpg (GnuPG) 2.0.26
libgcrypt 1.6.5
Tested with Firefox 47.0 and Crome 49.0.2623

The error-messages while import the key is "Unable to import key(s)! Internal error."

The debug-log is: 
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: /usr/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options -
-always-trust --homedir '/var/www/htdocs/roundcube/plugins/enigma/home/stefan.h' --version
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: BEGIN PROCESSING
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: selecting streams
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => got 1
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: GPG output stream ready for reading
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG output
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => read 616 bytes
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: selecting streams
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => got 3
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: GPG output stream ready for reading
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG output
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: GPG error stream ready for reading
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG error
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: GPG status stream ready for reading
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG status
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: END PROCESSING
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: CLOSING GPG SUBPROCESS
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => subprocess returned an unexpected exit code: -1
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: USING GPG 2.0.26 with PHP 5.6.21
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: OPENING GPG-AGENT SUBPROCESS WITH THE FOLLOWING COMMAND:
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: /usr/bin/gpg-agent --daemon --options /dev/null --csh --keep-display --no-grab --ignore-cache-for-signing --pi
nentry-touch-file /dev/null --disable-scdaemon --no-use-standard-socket --pinentry-program '/var/www/htdocs/roundcube/vendor/pear-pear.php.net/Crypt_GPG/Cry
pt/GPG/../../bin/crypt-gpg-pinentry' --homedir '/var/www/htdocs/roundcube/plugins/enigma/home/stefan.h'
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: CLOSING GPG-AGENT LAUNCH PROCESS
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: GPG-AGENT LAUNCH PROCESS CLOSED
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: /usr/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options -
-no-permission-warning --exit-on-status-write-error --trust-model always --with-colons --with-fingerprint --with-fingerprint --fixed-list-mode --homedir '/v
ar/www/htdocs/roundcube/plugins/enigma/home/stefan.h' --list-secret-keys
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: BEGIN PROCESSING
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: selecting streams
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => got 1
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: GPG error stream ready for reading
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG error
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => read 180 bytes
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: ERROR: gpg: keyring `/var/www/htdocs/roundcube/plugins/enigma/home/stefan.h/secring.gpg' created
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: ERROR: gpg: keyring `/var/www/htdocs/roundcube/plugins/enigma/home/stefan.h/pubring.gpg' created
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: selecting streams
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: GPG output stream ready for reading
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG output
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: GPG error stream ready for reading
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG error
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: GPG status stream ready for reading
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG status
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: END PROCESSING
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: CLOSING GPG SUBPROCESS
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: => subprocess returned an unexpected exit code: -1
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: STOPPING GPG-AGENT DAEMON
[23-Jun-2016 14:42:04 +0200]: <p8fsqkpc> GPG: GPG-AGENT DAEMON STOPPED
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: /usr/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options -
-always-trust --homedir '/var/www/htdocs/roundcube/plugins/enigma/home/stefan.h' --version
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: BEGIN PROCESSING
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: selecting streams
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => got 1
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG output stream ready for reading
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG output
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => read 616 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: selecting streams
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => got 3
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG output stream ready for reading
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG output
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG error stream ready for reading
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG error
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG status stream ready for reading
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG status
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: END PROCESSING
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: CLOSING GPG SUBPROCESS
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => subprocess returned an unexpected exit code: -1
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: USING GPG 2.0.26 with PHP 5.6.21
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: OPENING GPG-AGENT SUBPROCESS WITH THE FOLLOWING COMMAND:
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: /usr/bin/gpg-agent --daemon --options /dev/null --csh --keep-display --no-grab --ignore-cache-for-signing --pi
nentry-touch-file /dev/null --disable-scdaemon --no-use-standard-socket --pinentry-program '/var/www/htdocs/roundcube/vendor/pear-pear.php.net/Crypt_GPG/Cry
pt/GPG/../../bin/crypt-gpg-pinentry' --homedir '/var/www/htdocs/roundcube/plugins/enigma/home/stefan.h'
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: CLOSING GPG-AGENT LAUNCH PROCESS
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG-AGENT LAUNCH PROCESS CLOSED
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: OPENING GPG SUBPROCESS WITH THE FOLLOWING COMMAND:
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: /usr/bin/gpg --status-fd '3' --command-fd '4' --no-secmem-warning --no-tty --no-default-keyring --no-options -
-no-permission-warning --exit-on-status-write-error --trust-model always --homedir '/var/www/htdocs/roundcube/plugins/enigma/home/stefan.h' --import
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: BEGIN PROCESSING
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: selecting streams
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => got 1
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: input stream is ready for reading
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from input stream
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => read 8192 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: selecting streams
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => got 2
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG is ready for input
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to write 8192 bytes to GPG input
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => wrote 8192 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: input stream is ready for reading
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from input stream
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => read 2402 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: selecting streams
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => got 1
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG is ready for input
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to write 2402 bytes to GPG input
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => wrote 2402 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => closing GPG input pipe
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: selecting streams
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => got 1
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG error stream ready for reading
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG error
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => read 119 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: ERROR: gpg: Fatal: can't open `/var/www/htdocs/roundcube/plugins/enigma/home/stefan.h/trustdb.gpg': No such fi
le or directory
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => closing GPG input pipe
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: selecting streams
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => got 3
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG output stream ready for reading
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG output
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG error stream ready for reading
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG error
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG status stream ready for reading
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => about to read 8192 bytes from GPG status
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => read 0 bytes
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => closing GPG input pipe
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: END PROCESSING
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: CLOSING GPG SUBPROCESS
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: => subprocess returned an unexpected exit code: -1
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: STOPPING GPG-AGENT DAEMON
[23-Jun-2016 14:42:17 +0200]: <p8fsqkpc> GPG: GPG-AGENT DAEMON STOPPED

What can/must I do?

**Comment by alecpl** [comment]
@hestefan this is a different issue and it looks like a GnuPG bug. As I see it was fixed in 2.0.27.

**Comment by MetallianFR68** [comment]
Hi,
I'm expecting the exact same bug.
I spent 2 hours trying to solve it using solutions recommended here, including downloading the latest openpgp.js.
Nothing worked on CHROME 54
Then I tested with FIREFOX and it worked immediately.
It is a browser related problem

**Comment by NSkelsey** [comment]
@MetallianFR68 hey what version of Firefox did you run the test with?

**Comment by MetallianFR68** [comment]
@NSkelsey The latest : Firefox 49.0. I just installed it yesterday to test

**Comment by alecpl** [comment]
We've been asked for a sample key content with the issue. @MetallianFR68, @lutel, could you provide this information? Find https://github.com/roundcube/roundcubemail/blob/release-1.2/plugins/enigma/lib/enigma_ui.php#L501 and insert `rcube::console($data);` there. Then generate a new key in Chrome. Key content will be logged to logs/console file.

**Comment by MetallianFR68** [comment]
@alecpl Guess what ? today it's working fine under Chrome. I really don't understand why. I didn't change anything in my server and client settings !

**Comment by alecpl** [comment]
Clock issue described initially in this ticket is fixed in Crypt_GPG 1.6.0b3. As other issues here cannot be reliably reproduced, I'm closing this ticket.
